### ISSUE TITLE
CRM 2015 Toekn Issue

### ISSUE THREAD

**Issue by HajiJama** [Microsoft/CRM-Performance-Toolkit#25]
We wanted to load test CRM 2015 using visual studio load test but we ran into the known issue of the "WRPC Invalid Security Token" .  The only solution we found from Microsoft was to disable this token in the registry and run the load test which was successful.  The question we have is what is the CRM performance difference with or without this token enabled?  Microsoft  engineers who consulted with the Microsoft CRM folks could not answer this question clearly.  Did anyone do this comparison?  Also what is the difference of using the performance toolkit compared with using visual studio load test?

**Comment by irfanhusni** [comment]
hi @HajiJama do you have any solution from this problem? since i'm also have a same problem with you, many thanks.

**Comment by HajiJama** [comment]
Not really.  It seems you cannot load test CRM unless you disable the token
in the registry, according to Microsoft.  I read somewhere that you can
record the script in Fidler and upload to visual load test, but we did not
go there.

**Comment by CliveWilliamsNZ** [comment]
You need to approach this as a proper performance test.  The WRPC tokens are sent from the server and be correlated with the responses in a 'proper' performance testing tool.  I count Visual Studio Load test as a 'proper' one!
There are two issues here.  The first is that there is not a single WRPC token.  In a recent test, I counted 11 different ones being sent in a single response of which I had to pick the correct 2 to re-use later on in my script.  The other is I found that the encoding of the initial response is different from what needs to be sent in the request(s).  I had to write a custom decoder to handle.    
Anything you have to change in the configuration of the system under test potentially moves your testing away from reality.
